src/haddock/gear/prepare_run.py

```diff
@@ -517,3 +530,113 @@ def populate_mol_parameters(modules_params):
                 defaults[param],
                 )
     return
+
+
+def check_if_path_exists(path):
+    """
+    Check if a path exists and raises an error if it does not exist.
+
+    For example given this path "../config/project_01/file.txt" it would find
+    the following path "../config/project-01".
+
+    Parameters
+    ----------
+    path : AnyStr | PathLike
+        The path to check.
+
+    Returns
+    -------
+    None
+        If the path does exist.
+
+    Raises
+    ------
+    ValueError
+        If the path does not exist.
+    """
+    path = os.path.normpath(path)
+    if os.path.exists(path):
+        return None
+
+    split = os.path.split(path)
+    elements = split[0].split(os.sep) + split[1]
```

Comment:
Do you mind refactoring to using `pathlib` ? The community is moving towards using it always when handling paths.
```python
elements = Path(path).parts
for part in elements:
    next_folder = Path(reconstitutied_path, part)
    if not next_folder.exists():
         ....